okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -176,30 +177,40 @@ private static String switchWsToHttp(String url) {
         return redirectTo(current, httpUrls.get((currentUrl.get() + 1) % httpUrls.size()));
     }
 
+    @Override
+    public void markAsSucceeded(HttpUrl failedUrl) {
+        if (useFailedUrlCache) {
+            indexFor(failedUrl).ifPresent(failedUrls::invalidate);
```

Comment:
I wonder if O(1) -> O(n) here might be a problem since this happens on every successful request. I think at our current load though this will be negligible.

Reply:
Yeah this is not ideal, but I wasn't sure how else to get the desired behavior.